### DISCUSSION TITLE
Мои впечатления

### DISCUSSION THREAD

**Discussion by Gescha5555** [kachurovskiy/nanoels#178] [Ideas]
Здравствуйте Максим, меня зовут Геннадий, мне 56 лет.

Пользовался я гитарой от Олега уже 4 года, вроде работает нормально и всё устраивает, кроме скорости при нарезке резьбы. Но решил попробовать что-нибудь более современное, быстрое, точное на новой элементной базе. 

Решил попробовать вашу гитару, хотя отпугивало количество кнопок. По быстрому собрал (немного пришлось повозиться с контроллером клавиатуры – несколько микросхем с алиэкспресса оказались пробитые некоторые каналы), ну и приступил к тесту.

Понравилось:

* Скорость
* Точность
* Ввод данных
* Много видов обработки

Но есть и ложка дёгтя:

Максим, вы явно не токарь, логику работы я такой не видел. Ваша гитара всё больше начинает походить на станок с ЧПУ – это неплохо, но теряется главное достоинство – токарный универсальный станок.

Вот пример:

1. Мне надо нарезать резьбу.
2. Мои действия: выставляю упоры левые и правые, выставляю шаг, перемещаю в ручную поперечную подачу на нужную глубину и нажимаю кнопку пуск.
3. Станок режет резьбу до упора и останавливается.
4. Дальше начинается самое весёлое: отвожу поперечку, нажимаю и держу кнопку влево – суппорт перемещается до левого упора.
5. Теперь мне надо продолжать держать одной рукой кнопку, а другой подводить резец.
6. Если надо смазать, скинуть стружку – это очень не удобно.

Или этот же режим:

* Я протачиваю деталь.
* Потом мне надо торцануть.
* И понеслось: грязными руками начинаю 1) вкл режим торцовки, 2) выставить лимиты, 3) нажать пуск несколько раз (это 10 или больше нажатий разных кнопок).

Ещё по безопасности:

* У вас запуск одной кнопкой, выключение другой, управление третьей.
* Приходится переводить взгляд с зоны обработки на прибор, чтобы найти кнопку.
* А если к этому добавить большой станок?

Пример:

* Режим резьбу, шаг 3мм, длинной 15мм, вылет детали 25мм, обороты 400об.
* Запускаем пуск.
* Резец нарезает 15мм и тут мы видим, что упоры не выставили или что-то напутали.
* Успеете вы найти кнопку стоп, пока врежетесь в патрон? Я думаю нет.

**Как же здесь просто не хватает джойстика!**

Я сейчас думаю:

* Поставить параллельно кнопки влево кнопку с фиксацией, чтобы не держать кнопку.
* Ковырять код, чтобы сделать джойстик (хотя в программировании я не ас, так понемногу).

Фото моей сборки прилагаю. Большая просьба к вам: добавьте джойстик, и будет просто здорово!

С уважением, Геннадий

![IMG_20240212_200016_445](https://github.com/kachurovskiy/nanoels/assets/159788822/47d15efc-7cd7-494b-b33e-5a97cf880291)
![IMG_20240212_200101_945](https://github.com/kachurovskiy/nanoels/assets/159788822/4879883e-d9a7-4127-ad00-e86121376be8)

**Comment by kachurovskiy** [comment]
Геннадий, отлично сделано, красота, да еще и с переводом!

Если есть возможность, поделитесь пожалуйста своим модифицированным кодом - можно загрузить на https://gist.github.com/ и оставить ссылку тут.

Если есть электросхема и список компонентов вашего проекта то тоже поделитесь пожалуйста если можно.

> И ещё по безопасности у вас запуск одной кнопкой выключение другой управление третьей приходится переводить взгляд с зоны обработки на прибор ...  успеете вы найти кнопку стоп пока врежетесь в патрон я думаю нет

Отдельные кнопки на старт и стоп это как раз безопасней, посмотрите хоть магнитные пускатели на Али, везде старт и стоп раздельно. К тому же, кнопка аварийной остановки станка должна решать этот вопрос - если у вас такой нет, нужно добавить.

> Большая просьба к вам чтобы вы добавили джойстик ,и будет просто здорово.

Я так понял вы много пользуетесь режимом шестеренок? Может просто сделать чтобы при нажатии ![IconReverse](https://github.com/kachurovskiy/nanoels/assets/517919/d820c1cd-8ce3-4d02-91f9-d240f4ea51cc) резец оставалася в резьбе и тогда можно будет с помощью одного нажатия отправить резец назад и потом второго нажатия запустить следующий проход?

Добавление джойстика я одобряю, надеюсь что найдется умелец который выпустит версию с такой функцией.

**Reply by Gescha5555** [reply]
Режим шестерёнок это самый основной режим в нём работает все станки на99,5% времени , остальные режимы это приятное
 дополнение к основному режиму ,и насколько удобно и логично управление этим режимом будет зависеть
 станет ваша ELS народной и популярной . Я конечно понимаю что большинство людей которые купили
 маленькие токарные станочки не имеют представления и знания про режимы резанья про заточку резца
 про заточку свёрла я вообще молчу ,для них это дорогое баловство на котором точат какие нибудь ручки из
 пластика максимум что нибудь из алюминия .Люди которые действительно работают на станках ценят
 удобство управления ну и быстро настроиться .Самое логичное в управление станках
 когда нажал и держишь в право станок едет в право отпустил станок остановился и так же влево ,верх ,низ.
 Эта логика работы есть на всех современные да и на старых станках ,она отработана людьми веками
 и многие люди растались с жизнью или стали калеками пока всё отработали .
 Отсюда и про безопасность что я писал в первом сообщении нажал и держу - режу резьбу ,
 вижу  что резец едет дальше чем надо просто отпускаю палец это доли секунды .
 А у вас надо отвести взгляд найти кнопку стоп на панели или отдельную кнопку аварийной 
 остановки и нажать её как результат реэец в патроне (если конечно вы работаете одной рукой
 а второй всегда держитесь за аварийный выключатель).
 Я пробую на ней работать но все эти действия это много нажимать кнопки чтобы торцануть
 потом опять настраивать проточку просто отнимает время и убивает все достоинства вашей ELS.
 Поэтому гитару от Олега.А я далеко не убираю.
 Отсюда и просьба к вам подумайте всётки о изменении логики работы вашей ELS.
 У уважением Геннадий.

**Reply by kachurovskiy** [reply]
Геннадий, понял. Можете пожалуйста поделиться файлами своего проекта для желающих повторить? Спасибо.

**Reply by Gescha5555** [reply]
Максим я честно говоря незнаю чем мне поделиться
 схема полностью ваша , ESP32-S3 Dual Type-C полностью совподают пины ,
 контроллер клавиатуры распаян на переходнике ,преобразователи уровней 
 самые простые из китая ,кнопки накладки на кнопки дисплей всё из китая.
 И все компонента собирал на макетке ,пользовался вашей схемой там всё
 прекрасно распиновано я только распайку кнопок сделал как мне удобно.
 Прошивка полностью  ваша там только изменены коды кнопок ну и
 отопражение информации немного руссифицировал ,но ещё не закончил,
 я практически только допаял так что попозже как закончу.
 Так что могу пока только ссылки на AliExpress где брал всё рабочее.
 С уважением Геннадий.

**Reply by kachurovskiy** [reply]
Геннадий, ссылки на AliExpress будут очень полезны, так не совсем понятно что за компоненты использованы. Спасибо!

**Reply by Gescha5555** [reply]
ESP32-S3 Dual Type-C -
 https://aliexpress.ru/item/1005005051261468.html?spm=a2g2w.orderdetail.0.0.38b84aa6FUFZ2p&sku_id=12000031458955067

ЖК-дисплей 146*62,5 мм
https://aliexpress.ru/item/32827082989.html?spm=a2g2w.orderdetail.0.0.2bdf4aa6Oa1mpZ&sku_id=10000001419845786

Адаптер QFN24 для печатной платы (для монтажа TCA8418RTWR(PZ418))-
  https://aliexpress.ru/item/1005004865848522.html?spm=a2g2w.orderdetail.0.0.1a924aa6cJF1jH&sku_id=12000030805276178

Контроллер клавиатуры TCA8418RTWR(PZ418) ссылка больше не работает.

4-канальный преобразователь логического уровня IIC I2C 5v в 3.3v -
https://aliexpress.ru/item/32648070422.html?sku_id=12000037068993174&spm=a2g2w.productlist.search_results.9.286a5a3eQ8MYfK

Кнопки -
https://aliexpress.ru/item/1005001293076486.html?spm=a2g2w.orderdetail.0.0.42764aa6nexS0Y&sku_id=12000026796846581

Колпачки  на кнопки -
https://aliexpress.ru/item/1005001628106767.html?spm=a2g2w.orderdetail.0.0.22e44aa6sYY0Wt&sku_id=12000016886689642
 Это всё что заказывал ну и немножко резисторов да конденсаторов .

**Reply by Charus-OV** [reply]
TCA8418RTWR PZ418 
https://sl.aliexpress.ru/p?key=93NHOkD

**Comment by kotovasia5120** [comment]
@Gescha5555  у вас осталась модель панели для кнопок?

**Reply by Gescha5555** [reply]
Нет модели не осталось.
